src/ConnectionFactory.php

```diff
@@ -75,4 +83,31 @@ protected function getDefaultConfig($configKey)
             'doctrine_commented_types' => [],
         ];
     }
+
+    /**
+     * Registers all declared typed, if not already done.
+     *
+     * @param ContainerInterface $container
+     */
+    private function registerTypes(ContainerInterface $container)
+    {
+        if (self::$areTypesRegistered) {
+            return;
+        }
+
+        $applicationConfig = $container->has('config') ? $container->get('config') : [];
+        $doctrineConfig = array_key_exists('doctrine', $applicationConfig) ? $applicationConfig['doctrine'] : [];
+        $typesConfig = array_key_exists('types', $doctrineConfig) ? $doctrineConfig['types'] : [];
+
+        foreach ($typesConfig as $name => $className) {
+            if (Type::hasType($name)) {
+                Type::overrideType($name, $className);
+                continue;
+            }
+
+            Type::addType($name, $className);
+        }
+
+        self::$areTypesRegistered = true;
```

Comment:
Flag should be moved up, or else the logic may be re-executed if any of the `Type::*` operation throws an exception